src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ConstructorParametersShouldMatchPropertyAndFieldNames.cs

```diff
@@ -0,0 +1,215 @@
+                var memberWords = WordParser.Parse(referencedMember.Name, WordParserOptions.SplitCompoundWords);
+
+                return paramWords.SequenceEqual(memberWords, StringComparer.OrdinalIgnoreCase);
+            }
+
+            private bool IsJsonConstructor([NotNullWhen(returnValue: true)] IMethodSymbol? method)
+                => method.IsConstructor() &&
+                    method.HasAttribute(this._jsonConstructorAttributeInfoType);
+
+            private static IMemberReferenceOperation? TryGetMemberReferenceOperation(IParameterReferenceOperation paramOperation)
+            {
+                if (paramOperation.Parent is IAssignmentOperation assignmentOperation
+                    && assignmentOperation.Target is IMemberReferenceOperation assignmentTarget)
+                {
+                    return assignmentTarget;
+                }
+
+                if (paramOperation.Parent is ITupleOperation sourceTuple
+                    && sourceTuple.Parent is IConversionOperation conversion
+                    && conversion.Parent is IDeconstructionAssignmentOperation deconstruction
+                    && deconstruction.Target is ITupleOperation targetTuple)
+                {
+                    var paramIndexInTuple = sourceTuple.Elements.IndexOf(paramOperation);
+
+                    return targetTuple.Elements[paramIndexInTuple] as IMemberReferenceOperation;
+                }
+
+                return null;
+            }
+
+            private static void ReportFieldDiagnostic(OperationAnalysisContext context, DiagnosticDescriptor diagnosticDescriptor, ParameterDiagnosticReason reason, IParameterSymbol param, IFieldSymbol field)
+            {
+                var properties = ImmutableDictionary<string, string?>.Empty
+                    .SetItem(ReferencedFieldOrPropertyNameKey, field.Name)
+                    .SetItem(DiagnosticReasonKey, reason.ToString());
+
+                context.ReportDiagnostic(
+                    param.Locations[0].CreateDiagnostic(
+                        diagnosticDescriptor,
+                        reason == ParameterDiagnosticReason.FieldInappropriateVisibility ? field.Locations : ImmutableArray<Location>.Empty,
+                        properties,
+                        param.ContainingType.ToDisplayString(SymbolDisplayFormats.ShortSymbolDisplayFormat),
+                        param.Name,
+                        field.Name));
+            }
+
+            private static void ReportPropertyDiagnostic(OperationAnalysisContext context, DiagnosticDescriptor diagnosticDescriptor, ParameterDiagnosticReason reason, IParameterSymbol param, IPropertySymbol prop)
+            {
+                var properties = ImmutableDictionary<string, string?>.Empty
+                    .SetItem(ReferencedFieldOrPropertyNameKey, prop.Name)
+                    .SetItem(DiagnosticReasonKey, reason.ToString());
```

Comment:
Since we know the two keys being added are unique.

```suggestion
                    .Add(ReferencedFieldOrPropertyNameKey, prop.Name)
                    .Add(DiagnosticReasonKey, reason.ToString());
```

Reply:
Consider using constant strings instead of an enum, which avoids doing Enum.Parse in codefix, and also matches what's done in other analyzers.